splink/diagnostics.py

```diff
@@ -0,0 +1,201 @@
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.session import SparkSession
+from pyspark.context import SparkContext, SparkConf
+from pyspark.sql import SQLContext
+from pyspark.sql.types import StructType, StructField, StringType, DoubleType, Row
+import pyspark.sql.functions as f
+from pyspark.sql.functions import when
+from .check_types import check_types
+
+
+import pandas as pd
+import warnings
+
+
+altair_installed = True
+try:
+    import altair as alt
+except ImportError:
+    altair_installed = False
+
+
+hist_def_dict = {
+    "$schema": "https://vega.github.io/schema/vega-lite/v4.8.1.json",
+    "config": {
+        "title": {"fontSize": 14},
+        "view": {"continuousHeight": 300, "continuousWidth": 400},
+    },
+    "data": {"values": None},
+    "height": 200,
+    "mark": "bar",
+    "title": "Estimated Probability Density",
+    "width": 700,
+    "encoding": {
+        "tooltip": [{"field": "count_rows", "title": "count", "type": "quantitative"}],
+        "x": {
+            "axis": {"title": "splink score"},
+            "bin": "binned",
+            "field": "splink_score_bin_low",
+            "type": "quantitative",
+        },
+        "x2": {"field": "splink_score_bin_high"},
+        "y": {
+            "field": "normalised",
+            "type": "quantitative",
+            "axis": {"title": "Probability density"},
+        },
+    },
+}
+
+
+@check_types
+def _calc_probability_density(
+    df_e: DataFrame, spark: SparkSession, buckets=None, score_colname=None,
+):
+
+    """perform splink score histogram calculations / internal function 
+    
+    Compute a histogram using the provided buckets.
+    
+    
+        Args:
+    
+        
+            df_e (DataFrame): A dataframe of record comparisons containing a splink score, 
+            e.g. as produced by the .... function
+            
+            spark (SparkSession): SparkSession object
+            
+            score_colname : is the score in another column? defaults to None
+            
+            buckets : accepts either a list of split points or an integer number that is used to create equally spaced split points. 
+            It defaults to 100 equally spaced split points from 0.0 to 1.0
+            
+        
+        Returns:
+            
+            (DataFrame) : pandas dataframe of histogram bins for appropriate splink score variable ready to be plotted.
+            
+        
+    """
+
+    # if splits a list then use it. if None... then create default. if integer then create equal bins
+
+    if isinstance(buckets, int) and buckets != 0:
```

Comment:
`rdd.histogram` accepts an `int` or ` list of buckets, so this logic isn't needed (i.e. `rdd.histogram` will do the same thing as t his logic!)